### DISCUSSION TITLE
游戏全屏独占和桌面分辨率不同为何会导致切换桌面黑屏？

### DISCUSSION THREAD

**Discussion by mailliangzi** [Blinue/Magpie#1018] [General]
而有些游戏则可以低于桌面分辨率全屏。有些则不可以，例如cs2在全屏模式若使用低于桌面分辨率的情况就会导致切换桌面会黑屏且 系统ui会错位重新刷新

**Comment by hooke007** [comment]
你这个问题直接问百度/bing更快。。。

**Comment by Blinue** [comment]
这涉及到独占全屏和无边框全屏的区别。独占全屏时，游戏完全接管了桌面，黑屏和图标错位是因为显示器分辨率被游戏改了；无边框全屏时桌面依然由 dwm 绘制，游戏只是一个普通窗口。

传统上认为独占全屏的性能更好（因为不再渲染桌面和其他窗口），延迟更低（因为画面直接输出到显示器无需由 dwm 合成，但这个优势由于 DirectFlip 已经不存在了），无边框全屏的优势则在于和桌面的无缝集成，比如可以快速切换窗口，不会黑屏。
